Config/FilterInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Config;
+
+interface FilterInterface
+{
+    /**
+     * @return string
+     */
+    public function getName(): string;
+
+    /**
+     * @return array
+     */
+    public function getOptions(): array;
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void;
```

Comment:
imo the interface should not define setters. the whole definition is a read-only thing.

Reply:
completely agree with you :)